core/src/main/resources/org/apache/spark/ui/static/executorspage.js

```diff
@@ -25,6 +25,14 @@ function getThreadDumpEnabled() {
     return threadDumpEnabled;
 }
 
+function formatLossReason(status, type, row) {
```

Comment:
I see how you came up with the function arguments :)

The first argument must be the data which is selected from the whole REST response by the `data` field:
https://github.com/apache/spark/blob/433ee83861a0a3aa7aea3c05d02fc9e70579ce76/core/src/main/resources/org/apache/spark/ui/static/executorspage.js#L548

So please rename`status` to `removeReason` as we should format remove reasons.

And I guess its value was always `null`. This is why you used the `row` to access it. The root cause of this problem is the selection itself was wrong. 

So instead of 
```
 data: 'executorLossReason',  
```
it must be:
```
 data: 'removeReason', 
 ```
 
 And after this you can use the `removeReason` argument of this `formatLossReason` function to get the value.

Reply:
ah this makes sense. Thanks, that'll simplify the code :)